lib/plug_datadog_stats.ex

```diff
@@ -1,43 +1,62 @@
 defmodule PlugDatadogStats do
   @behaviour Plug
-  import Plug.Conn, only: [register_before_send: 2]
+  import Plug.Conn, only: [register_before_send: 2, put_private: 3]
   require Logger
 
-  def init(_) do
-    %{
-      histogram: Application.get_env(:plug_datadog_stats, :histogram_name, "resp_time"),
-      count: Application.get_env(:plug_datadog_stats, :count_name, "resp_count"),
-    }
+  @base_opts %{
+    histogram: Application.get_env(:plug_datadog_stats, :histogram_name, "resp_time"),
+    count: Application.get_env(:plug_datadog_stats, :count_name, "resp_count"),
+    path_normalize_fn: {__MODULE__, :path_override_passthrough, []}
+  }
+
+  def init(opts \\ []) do
+    opts
+    |> Enum.into(@base_opts)
+    |> parse_options()
   end
 
-  def call(conn, names) do
+  def call(conn, opts) do
     req_start_time = :os.timestamp
 
     register_before_send conn, fn conn ->
-      tags = tags_for_conn(conn)
 
       # This will log response time in microseconds
       req_end_time = :os.timestamp
       duration = :timer.now_diff(req_end_time, req_start_time)
 
-      Logger.debug("PlugDatadogStats: #{duration}µs #{names.histogram}/#{names.count} #{inspect tags}")
-      ExStatsD.histogram(duration, names.histogram, tags: tags)
-      ExStatsD.increment(names.count, tags: tags)
+      tags = tags_for_conn(conn, opts.path_normalize_fn)
+
+      Logger.debug("PlugDatadogStats: #{duration}µs #{opts.histogram}/#{opts.count} #{inspect tags}")
+      ExStatsD.histogram(duration, opts.histogram, tags: tags)
+      ExStatsD.increment(opts.count, tags: tags)
 
-      conn
+      put_private(conn, :plug_datadog_stats_tags, tags)
```

Comment:
I realize this is a little bit weird, but it allows me to actually test this module